sdk/nodejs/tests/runtime/closure-integreation-tests.ts

```diff
@@ -0,0 +1,115 @@
+    const { stdout } = await execa("npm", ["pack"], { cwd: dir });
+    const tarball = stdout.trim();
+    const tarballPath = path.join(dir, tarball);
+    await fs.rename(tarballPath, out);
+}
+
+// Write a package.json that installs the local pulumi package and ensures we
+// have the right typescript version.
+async function writePackageJSON(
+    dir: string,
+    pulumiPackagePath: string,
+    typescriptVersion: string,
+    nodeTypesVersion: string,
+) {
+    const packageJSON = {
+        name: "closure-tests",
+        version: "1.0.0",
+        license: "Apache-2.0",
+        dependencies: {
+            "@pulumi/pulumi": pulumiPackagePath,
+            "@types/mocha": "^9.0.0",
+            "@types/node": nodeTypesVersion,
+            "@types/semver": "^7.5.6",
+            mocha: "^9.0.0",
+            "mocha-suppress-logs": "^0.5.1",
+            mockpackage: "file:mockpackage",
+            semver: "^7.5.4",
+            "ts-node": "^7.0.1",
+            typescript: typescriptVersion,
+        },
+        resolve: {
+            typescript: typescriptVersion,
+        },
+    };
+
+    await fs.writeFile(path.join(dir, "package.json"), JSON.stringify(packageJSON, undefined, 4));
+}
+
+async function copyDir(src: string, dest: string) {
+    const entries: any[] = await fs.readdir(
+        src,
+        { recursive: true, withFileTypes: true } as any /* recursive option is new in node 18 */,
+    );
+
+    for (const entry of entries) {
+        const srcPath = path.join(entry.path, entry.name);
+        const destPath = srcPath.replace(src, dest);
+        const destDir = path.dirname(destPath);
+
+        if (entry.isFile()) {
+            await fs.mkdir(destDir, { recursive: true });
+            await fs.copyFile(srcPath, destPath);
+        }
+    }
+}
+
+async function run(typescriptVersion: string, nodeTypesVersion: string) {
+    const tmpDir = tmp.dirSync({ prefix: "closure-test-", unsafeCleanup: true });
+    const sdkRoot = path.join(__dirname, "..", "..", "..");
+    const sdkRootBin = path.join(sdkRoot, "bin");
+    // Add a random suffix to the package name to avoid any issues with yarn caching the tgz.
+    const packageName = `pulumi-${randomInt(10000, 99999)}.tgz`;
+    const pulumiPackagePath = path.join(tmpDir.name, packageName);
+    await pack(sdkRootBin, pulumiPackagePath);
+    await writePackageJSON(tmpDir.name, pulumiPackagePath, typescriptVersion, nodeTypesVersion);
+    await copyDir(path.join(sdkRoot, "tests", "runtime", "testdata", "closure-tests"), tmpDir.name);
+
+    await execa("yarn", ["install"], { cwd: tmpDir.name });
+
+    await execa("./node_modules/.bin/tsc", { cwd: tmpDir.name });
+
+    await execa("node", ["./node_modules/.bin/mocha", "--timeout", "30000", "test.js"], {
+        cwd: tmpDir.name,
+        stdio: "inherit",
+    });
+
+    tmpDir.removeCallback();
+}
+
+async function main() {
+    for (const [ts, typesNode] of [
+        ["~3.8.3", "^17"],
+        // ["^4.9.5", "^20"],
```

Comment:
In prep for supporting the other versions?

Reply:
Yes, in the follow up PR I make use of this here https://github.com/pulumi/pulumi/pull/15761/files#diff-641029ea7bb84996df592ea47b810b71ab72bedf625235a56a24a438b5fa4b64

I can tidy up the PR here if you prefer.